langchain/src/retrievers/self_query/utils.ts

```diff
@@ -16,3 +21,184 @@ export function isFilterEmpty(
+        if (typeof target[prop as SupabaseFilterProps] === "function") {
+          return (...args: any[]) => {
+            if (this.duplicationAllowedOps.includes(String(prop))) {
+              if (String(prop) === "or") {
+                const filters = args[0] as string;
+                const { foreignTable } =
+                  args[1] ?? ({} as { foreignTable?: string });
+                this.or(filters, { foreignTable });
+              } else if (String(prop) === "filter") {
+                const column = args[0] as string;
+                const operator = args[1] as string;
+                const value = args[2] as unknown;
+                this.filter(column, operator, value);
+              } else if (String(prop) === "in") {
+                const column = args[0] as string;
+                const values = args[1] as unknown[];
+                this.in(column, values);
+              } else if (String(prop) === "contains") {
+                const column = args[0] as string;
+                const value = args[1] as unknown[];
+                this.contains(column, value);
+              } else if (String(prop) === "textSearch") {
+                const column = args[0] as string;
+                const query = args[1] as string[];
+                const { config, type } = (args[2] ?? {}) as {
+                  config?: string;
+                  type?: "plain" | "phrase" | "websearch";
+                };
+                this.textSearch(column, query, { config, type });
+              } else if (String(prop) === "match") {
+                const query = args[0] as Record<string, unknown>;
+                this.match(query);
+              } else if (String(prop) === "not") {
+                const column = args[0] as string;
+                const operator = args[1] as string;
+                const value = args[2] as unknown;
+                this.not(column, operator, value);
+              } else {
+                const column = args[0] as string;
+                const value = args[1] as string;
+                this.defaultOp(prop as string, column, value);
+              }
+              return new Proxy(target, proxyHandler);
+            } else {
+              throw new Error(
+                "Duplication operation not supported for 'or' mergeFiltersOperator"
+              );
+            }
+          };
+        } else {
+          return Reflect.get(target, prop, receiver);
+        }
+      },
+    };
+
+    return proxyHandler;
+  }
+
+  addToValues(column: string, value: string) {
+    this.values.push([column, value]);
+  }
+
+  removeType(value: string) {
+    if (value.includes("::int")) {
+      return value.replace("::int", "");
+    }
+    return value;
+  }
+
+  defaultOp(prop: string, column: string, value: unknown) {
+    this.values.push([this.removeType(column), `${String(prop)}.${value}`]);
+  }
+
+  or(filters: string, { foreignTable }: { foreignTable?: string } = {}) {
```

Comment:
Can we rename these to `addOrClause` or something similar?

Just not obvious above that `this.or()` above has side-effects

Reply:
Hmm let me rename the rest too to make it consistent.